Sources/StreamChat/Workers/ChatClientUpdater.swift

```diff
@@ -159,7 +159,7 @@ class ChatClientUpdater {
             return
         }
 
-        guard client.connectionId != nil else {
+        if client.connectionId == nil && source == .userInitiated {
```

Comment:
I'm wondering if going past this expression for system initiated calls and calling `client.completeConnectionIdWaiters(connectionId: nil)` might cause any unpleasant side-effects, but probably it's all right

Reply:
you are 100% right, fixing this